packages/core/src/uint8array.ts

```
@@ -0,0 +1,554 @@
+): string {
+    assertString(string);
+    return uint8ArrayToBase64(stringToUint8Array(string), options);
+}
+
+const byteToHexLookupTable = Array.from({ length: 256 }, (_, index) =>
+    index.toString(16).padStart(2, '0'),
+);
+
+export function uint8ArrayToHex(array: Uint8Array): string {
+    assertUint8Array(array);
+
+    // Concatenating a string is faster than using an array.
+    let hexString = '';
+
+    // Max performance is critical, using for loop instead of array methods
+    for (let index = 0; index < array.length; index++) {
+        hexString += byteToHexLookupTable[array[index]];
+    }
+
+    return hexString;
+}
+
+const hexToDecimalLookupTable: Record<string, number | undefined> = {
+    0: 0,
+    1: 1,
+    2: 2,
+    3: 3,
+    4: 4,
+    5: 5,
+    6: 6,
+    7: 7,
+    8: 8,
+    9: 9,
+    a: 10,
+    b: 11,
+    c: 12,
+    d: 13,
+    e: 14,
+    f: 15,
+    A: 10,
+    B: 11,
+    C: 12,
+    D: 13,
+    E: 14,
+    F: 15,
+};
+
+/**
+ * Convert a hex string to a Uint8Array.
+ * Returns empty Uint8Array for invalid input (lenient behavior).
+ */
+export function hexToUint8Array(hexString: string): Uint8Array {
+    assertString(hexString);
+
+    // Return empty array for empty string or invalid length
+    if (hexString.length === 0 || hexString.length % 2 !== 0) {
+        return new Uint8Array(0);
+    }
+
+    const resultLength = hexString.length / 2;
+    const bytes = new Uint8Array(resultLength);
+
+    for (let index = 0; index < resultLength; index++) {
+        const highNibble = hexToDecimalLookupTable[hexString[index * 2]];
+        const lowNibble = hexToDecimalLookupTable[hexString[index * 2 + 1]];
+
+        // Return empty array for invalid hex characters (lenient)
+        if (highNibble === undefined || lowNibble === undefined) {
+            return new Uint8Array(0);
+        }
+
+        bytes[index] = (highNibble << 4) | lowNibble; // eslint-disable-line no-bitwise
+    }
+
+    return bytes;
+}
```

should throw an error.
also update the doc string.

```suggestion
export function hexToUint8Array(hexString: string): Uint8Array {
    assertString(hexString);

    // Return empty array for empty string or invalid length
    if (hexString.length === 0 || hexString.length % 2 !== 0) {
        throw new Error('Invalid hex string');
    }

    const resultLength = hexString.length / 2;
    const bytes = new Uint8Array(resultLength);

    for (let index = 0; index < resultLength; index++) {
        const highNibble = hexToDecimalLookupTable[hexString[index * 2]];
        const lowNibble = hexToDecimalLookupTable[hexString[index * 2 + 1]];

        // Return empty array for invalid hex characters (lenient)
        if (highNibble === undefined || lowNibble === undefined) {
            throw new Error('Invalid hex string');
        }

        bytes[index] = (highNibble << 4) | lowNibble; // eslint-disable-line no-bitwise
    }

    return bytes;
}
```